File: compiled_lang/cs/src/b20.formatting.cs
using System;

class B20Formatting
{
    static void Main()
    {
        int number = 5;
        char character = 'a';
        string text = "This is a string";

        // Positional format arguments - {0}/{1}/{2} bound to argument
        //  index, C#'s closest analog to printf's ordinal verbs (contrast
        //  with b10's "{number}" interpolation).
        Console.WriteLine(string.Format("Number is {0}.\nCharacter is '{1}'.\nString is \"{2}\".", number, character, text));
    }
}

File: lessons/compiled_lang/rust/src/b20.formatting.rs
fn main() {
    let number = 5;
    let character = 'a';
    let text = "This is a string";

    // Positional format arguments - Rust's closest std equivalent to
    //  printf's ordinal %1$d-style verbs, each `{N}` bound to an argument
    //  index rather than a captured variable name (contrast with b10's
    //  "{number}" interpolation).
    print!("Number is {0}.\nCharacter is '{1}'.\nString is \"{2}\".\n", number, character, text);
}
